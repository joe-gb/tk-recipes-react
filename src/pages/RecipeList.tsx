import RecipeService, { Recipe } from "../services/Recipe";
import { Link } from "react-router-dom";
import React, { useState } from "react";
import { Input } from "../components/Input";
import { Subtitle } from "../components/Text";
import { LinkList } from "../components/List";
import Loader from "../components/Loader";

function RecipeSearch() {
  const [search, setSearch] = useState("");
  let [recipes, loading, error] = RecipeService.GetRecipes(search);

  if (error) {
    return <p>Unable to fetch recipes - {error.message}</p>;
  }

  if (loading) {
    return <Loader />;
  }

  if (recipes.length === 0) {
    return (
      <p>
        No recipes match this criteria, why not{" "}
        <Link to="/recipes/add">create one?</Link>
      </p>
    );
  }
  return (
    <>
      <Subtitle>Recipe Search</Subtitle>
      <div>
        <Input
          placeholder="Search recipes"
          type="text"
          value={search}
          onChange={(event) => setSearch(event.target.value)}
        />
        <div>
          {LinkList(
            recipes.map((recipe: Recipe) => {
              return {
                link: "/recipes/" + recipe.id,
                text: recipe.name,
              };
            })
          )}
        </div>
      </div>
    </>
  );
}

export default RecipeSearch;
