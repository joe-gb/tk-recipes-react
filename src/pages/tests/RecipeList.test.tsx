import React from "react";
import axios from "axios";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import MockAdapter from "axios-mock-adapter";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, waitFor, render, screen } from "@testing-library/react";
import RecipeList from "../RecipeList";

const mock = new MockAdapter(axios);

mock.onGet(/http:\/\/localhost:8000\/recipes\/\?[0-9]+/).reply(200, [
  { id: 1, name: "Paella" },
  { id: 2, name: "Pizza" },
  { id: 3, name: "Sangria" },
]);

afterEach(cleanup);

test("Test recipe list", async () => {
  const { container } = render(
    <Router>
      <RecipeList />
    </Router>
  );

  await waitFor(() => {
    expect(screen.getByText("Paella")).toBeInTheDocument();
  });
  expect(screen.getByText("Pizza")).toBeInTheDocument();
  expect(screen.getByText("Sangria")).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
