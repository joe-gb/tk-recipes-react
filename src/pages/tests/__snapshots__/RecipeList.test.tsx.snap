// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test recipe list 1`] = `
.c0 {
  font-size: 32px;
  font-weight: 500;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  line-height: 40px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #1d2c3c;
  margin: 20px auto;
  max-width: 850px;
}

.c1 {
  margin: 10px auto;
}

.c1 div {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  min-height: 40px;
  z-index: initial;
  font-size: 16px;
  background: #ffffff;
  caret-color: auto;
  border: 1px solid #C9D3DF;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  -webkit-transition: opacity 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out,border-color 0.2s ease-in-out;
}

.c1 div div:hover {
  border-left: 1px solid #738094;
  border-right: 1px solid #738094;
  border-top: 1px solid #738094;
  border-bottom: 1px solid #738094;
}

.c1 div input {
  height: 40px;
  box-sizing: border-box;
  width: 100%;
  padding: 0 14px;
  border-radius: 2px;
  line-height: 32px;
  font-size: 16px;
  border: none;
  color: #323b4d;
}

.c1 div input:focus {
  outline: none;
}

.c2 {
  font-size: 18px;
  font-weight: 400;
  line-height: 24px;
  text-align: left;
  vertical-align: baseline;
  text-transform: unset;
  color: #4c5566;
  background-color: #eff2f7;
  margin: 2px 0px 2px 0px;
  padding: 10px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <h2
    class="c0"
  >
    Recipe Search
  </h2>
  <div>
    <div
      class="c1"
    >
      <div>
        <input
          id="sc-iCfMLu c1"
          placeholder="Search recipes"
          type="text"
          value=""
        />
      </div>
    </div>
    <div>
      <a
        href="/recipes/1"
      >
        <div
          class="c2"
        >
          Paella
        </div>
      </a>
      <a
        href="/recipes/2"
      >
        <div
          class="c2"
        >
          Pizza
        </div>
      </a>
      <a
        href="/recipes/3"
      >
        <div
          class="c2"
        >
          Sangria
        </div>
      </a>
    </div>
  </div>
</div>
`;
